src/js/game/hud/parts/debug_info.js

```diff
@@ -2,28 +2,66 @@ import { BaseHUDPart } from "../base_hud_part";
 import { makeDiv, round3Digits, round2Digits } from "../../../core/utils";
 import { DynamicDomAttach } from "../dynamic_dom_attach";
 import { KEYMAPPINGS } from "../../key_action_mapper";
+import { Vector } from "../../../core/vector";
+
+/** @enum {string} */
+export const enumDebugOverlayMode = { disabled: "disabled", regular: "regular", detailed: "detailed" };
 
 export class HUDDebugInfo extends BaseHUDPart {
     createElements(parent) {
         this.element = makeDiv(parent, "ingame_HUD_DebugInfo", []);
 
         this.tickRateElement = makeDiv(this.element, null, ["tickRate"], "Ticks /s: 120");
         this.fpsElement = makeDiv(this.element, null, ["fps"], "FPS: 60");
-        this.tickDurationElement = makeDiv(this.element, null, ["tickDuration"], "Update time: 0.5ms");
+        this.tickDurationElement = makeDiv(this.element, null, ["tickDuration"], "Tick dur: 0.5ms");
+        this.mousePositionElement = makeDiv(this.element, null, ["mousePosition"], "Pos: 0 0");
+        this.cameraPositionElement = makeDiv(this.element, null, ["cameraPosition"], "Center: 0 0");
+        this.versionElement = makeDiv(this.element, null, ["version"], "version unknown");
     }
 
     initialize() {
         this.lastTick = 0;
 
-        this.visible = false;
+        this.mode = enumDebugOverlayMode.disabled;
         this.domAttach = new DynamicDomAttach(this.root, this.element);
 
         this.root.keyMapper.getBinding(KEYMAPPINGS.ingame.toggleFPSInfo).add(() => this.toggle());
     }
 
+    updateFullText() {
+        this.element.classList.toggle("debugFull", this.full);
```

Comment:
If you want to do it very clean, you could use `.setAttribute("data-mode", this.mode)`.

Also this.full is undefined